Use params for log messages

matt-richardson opened the pull request:
The logging that comes out of `WebJobs.Extensions.*` mostly uses params for structured logging, but some spots just use plain old interpolated strings.

This means that it's hard to filter these out, as they're all different:

![image](https://github.com/Azure/azure-webjobs-sdk-extensions/assets/373389/7b8ba6ba-1b9f-4f7a-83e7-e1840d33984b)

This PR updates the log messges so they use params better, like structured logs should be.

matt-richardson commented:
@jviau - I see you're active in this repo at the moment.

Any chance we could get this PR in?

jviau commented on src/WebJobs.Extensions/Extensions/Timers/Listener/TimerListener.cs:
```diff
@@ -342,7 +342,8 @@ internal async Task InvokeJobFunction(DateTime invocationTime, bool isPastDue =
                 {
                     await ScheduleMonitor.UpdateStatusAsync(_timerLookupName, ScheduleStatus);
-                    _logger.LogDebug($"Function '{_functionLogName}' updated status: Last='{ScheduleStatus.Last.ToString("o")}', Next='{ScheduleStatus.Next.ToString("o")}', LastUpdated='{ScheduleStatus.LastUpdated.ToString("o")}'");
+                    _logger.LogDebug("Function '{functionName}' updated status: Last='{lastScheduled}\', Next=\'{nextScheduled}\', LastUpdated=\'{lastUpdated}\'", 
```
nit, use formatter syntax:

```suggestion
                    _logger.LogDebug("Function '{functionName}' updated status: Last='{lastScheduled:o}\', Next=\'{nextScheduled:o}\', LastUpdated=\'{lastUpdated:o}\'", 
```

And then you can remove the `.ToString("o")` calls

jviau commented:
/azp run webjobs-sdk-extensions.public

jviau commented:
@microsoft-github-policy-service rerun

jviau commented:
@matt-richardson if you can address the merge conflict and the formatting suggestion I will merge this in. Thank you for your contribution!

matt-richardson commented:
> @matt-richardson if you can address the merge conflict and the formatting suggestion I will merge this in. Thank you for your contribution!

Sorry, somehow missed your message.

I've fixed the merge conflict, and incorporated your suggestion.

Unfortunately, my IDE got over-enthusiastic and trimmed some trailing spaces. Let me know if you need me to undo that change.

jviau commented on src/WebJobs.Extensions/Extensions/Timers/Listener/TimerListener.cs:
```diff
@@ -402,7 +402,8 @@ internal async Task InvokeJobFunction(DateTimeOffset invocationTime, bool isPast
                 {
                     await ScheduleMonitor.UpdateStatusAsync(_timerLookupName, ScheduleStatus);
-                    _logger.LogDebug($"Function '{_functionLogName}' updated status: Last='{ScheduleStatus.Last.ToString("o")}', Next='{ScheduleStatus.Next.ToString("o")}', LastUpdated='{ScheduleStatus.LastUpdated.ToString("o")}'");
+                    _logger.LogDebug("Function '{functionName}' updated status: Last='{lastScheduled:o}', Next='{nextScheduled:o}', LastUpdated='{lastUpdated:p}'",
```
Possible typo? You have `lastUpdated:p` , I think it should be `:o` (o, not p)

```suggestion
                    _logger.LogDebug("Function '{functionName}' updated status: Last='{lastScheduled:o}', Next='{nextScheduled:o}', LastUpdated='{lastUpdated:o}'",
```

matt-richardson replied:
🤦

I'm glad you're paying attention!

jviau commented:
> Unfortunately, my IDE got over-enthusiastic and trimmed some trailing spaces.

That is always a welcome change 😄

jviau commented:
/azp run webjobs-sdk-extensions.public

jviau commented:
Thank you for your contribution @matt-richardson
